extensions/wikidata/module/scripts/dialogs/manage-account-dialog.js

```diff
@@ -2,120 +2,210 @@ var ManageAccountDialog = {};
 
 ManageAccountDialog.firstLogin = true;
 
-ManageAccountDialog.launch = function(logged_in_username, callback) {
-   $.post(
-      "command/core/get-all-preferences",
-      null,
-      function (preferences) {
-         ManageAccountDialog.display(logged_in_username, preferences.wikidata_credentials, callback);
-      },
-      "json"
-  );
-};
+ManageAccountDialog.display = function (logged_in_username, callback) {
 
-ManageAccountDialog.display = function(logged_in_username, saved_credentials, callback) {
-  var self = this;
-  var frame = $(DOM.loadHTML("wikidata", "scripts/dialogs/manage-account-dialog.html"));
-  var elmts = this._elmts = DOM.bind(frame);
-
-  this._elmts.dialogHeader.text($.i18n('wikidata-account/dialog-header'));
-  this._elmts.explainLogIn.html($.i18n('wikidata-account/explain-log-in'));
-  this._elmts.usernameLabel.text($.i18n('wikidata-account/username-label'));
-  this._elmts.usernameInput.attr("placeholder", $.i18n('wikidata-account/username-placeholder'));
-  this._elmts.passwordLabel.text($.i18n('wikidata-account/password-label'));
-  this._elmts.passwordInput.attr("placeholder", $.i18n('wikidata-account/password-placeholder'));
-  this._elmts.rememberCredentialsLabel.text($.i18n('wikidata-account/remember-credentials-label'));
-  this._elmts.dialogHeader.text($.i18n('wikidata-account/dialog-header'));
-  this._elmts.cancelButton1.text($.i18n('wikidata-account/close'));
-  this._elmts.cancelButton2.text($.i18n('wikidata-account/close'));
-  this._elmts.loggedInAs.text($.i18n('wikidata-account/logged-in-as'));
-  this._elmts.logoutButton.text($.i18n('wikidata-account/log-out'));
-  this._elmts.loginButton.val($.i18n('wikidata-account/log-in'));
+  if (logged_in_username == null) {
+    logged_in_username = ManageAccountDialog.tryLoginWithCookies(callback);
+  }
 
   if (logged_in_username != null) {
-      elmts.loginArea.remove();
+    ManageAccountDialog.displayLoggedIn(logged_in_username, callback);
   } else {
-      elmts.logoutArea.remove();
+    ManageAccountDialog.displayPasswordLogin(callback);
   }
+};
+
+ManageAccountDialog.tryLoginWithCookies = function (callback) {
+  var logged_user_name = null;
+  $.ajaxSetup({async: false});
```

Comment:
This is going to block all interactivity. Normally it should return quickly, but it would be better to do it asynchronously and allow the user to cancel the operation if it's taking too long.

I know there are a bunch of other places that this is done in OpenRefine, so it's natural that you'd follow the pattern. 

Since this is new code, perhaps we should consider using Promises. They provide a much cleaner pattern.

Reply:
I fixed this directly in https://github.com/OpenRefine/OpenRefine/tree/wikidata-extension-oauth.

Reply:
Thank you!